packages/webviz-core/src/dataProviders/MemoryCacheDataProvider.js

```diff
@@ -84,14 +89,35 @@ export function getBlocksToKeep({
         return {
           blockIndexesToKeep,
           // Adjust the oldest `newRecentRanges`.
-          newRecentRanges: [...recentBlockRanges.slice(0, blockRangeIndex), { start: blockIndex, end: blockRange.end }],
+          newRecentRanges: [
+            ...recentBlockRanges.slice(0, blockRangeIndex),
+            { start: Math.min(blockIndex, startIndex), end: Math.max(blockIndex, startIndex) + 1 },
```

Comment:
This looks correct, but it took me a few minutes to wrap my head around it. Maybe clearer would be something like `increment > 0 ? { … } : { … }`?

Reply:
You're right, that's much better.